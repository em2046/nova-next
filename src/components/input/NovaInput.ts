import { defineComponent, h, VNode } from 'vue'

export default defineComponent({
  setup() {
    return (): VNode => {
      return h('div', [h('input', { class: 'nova-input', type: 'text' })])
    }
  },
})
