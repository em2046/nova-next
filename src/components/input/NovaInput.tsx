import { computed, defineComponent } from 'vue';
import { vueJsxCompat } from '../../vue-jsx-compat';
import useEnvironment, {
  EnvironmentProps,
  environmentProps,
} from '../../uses/use-environment';

export default defineComponent({
  name: 'NovaInput',
  inheritAttrs: false,
  props: {
    ...environmentProps,
    class: {
      type: [String, Array, Object],
      default: null,
    },
    wrapClass: {
      type: [String, Array, Object],
      default: null,
    },
    wrapStyle: {
      type: Object,
      default: null,
    },
    disabled: {
      type: Boolean,
      default: false,
    },
    readonly: {
      type: Boolean,
      default: false,
    },
  },
  setup(props, context) {
    const environment = useEnvironment(props as EnvironmentProps);

    const wrapClassList = computed(() => {
      return [
        {
          'nova-input': true,
          'nova-input-disabled': !!props.disabled,
          'nova-input-readonly': !!props.readonly,
        },
        props.wrapClass,
      ];
    });

    const classList = computed(() => {
      return ['nova-input-text', props.class];
    });

    return (): JSX.Element => {
      return (
        <div
          class={wrapClassList.value}
          style={props.wrapStyle}
          data-nova-theme={environment.themeRef.value}
        >
          <input
            type="text"
            class={classList.value}
            {...context.attrs}
            disabled={!!props.disabled}
            readonly={!!props.readonly}
          />
          <div class="nova-input-border" />
        </div>
      );
    };
  },
});
