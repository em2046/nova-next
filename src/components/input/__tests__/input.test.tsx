import { mount } from '@vue/test-utils';
import { vueJsxCompat } from '../../../vue-jsx-compat';
import { NovaInput } from '../NovaInput';

describe('input', () => {
  test('render', () => {
    const wrapper = mount({
      setup() {
        return () => {
          return (
            <div>
              <NovaInput />
              <NovaInput disabled />
              <NovaInput readonly />
            </div>
          );
        };
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
