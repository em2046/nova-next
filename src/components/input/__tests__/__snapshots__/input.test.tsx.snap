// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input render 1`] = `"<div><div class=\\"nova-input\\" data-nova-theme=\\"light\\"><input type=\\"text\\" class=\\"nova-input-text\\"><div class=\\"nova-input-border\\"></div></div><div class=\\"nova-input nova-input-disabled\\" data-nova-theme=\\"light\\"><input type=\\"text\\" class=\\"nova-input-text\\" disabled=\\"\\"><div class=\\"nova-input-border\\"></div></div><div class=\\"nova-input nova-input-readonly\\" data-nova-theme=\\"light\\"><input type=\\"text\\" class=\\"nova-input-text\\" readonly=\\"\\"><div class=\\"nova-input-border\\"></div></div></div>"`;
